edb/server/dbview/dbview.pyx

```diff
@@ -148,6 +163,141 @@ cdef class Database:
+                await asyncio.sleep(0.1)
+
+    async def _cache_worker(self):
+        while True:
+            # First, handle any evictions
+            keys = []
+            while self._eql_to_compiled.needs_cleanup():
+                query_req, unit_group = self._eql_to_compiled.cleanup_one()
+                if len(unit_group) == 1 and unit_group.cache_state == 1:
+                    keys.append(query_req.get_cache_key())
+                unit_group.cache_state = CacheState.Evicted
+            if keys:
+                await self.tenant.evict_query_cache(self.name, keys)
+
+            # Now, populate the cache
+            # Empty the queue, for batching reasons.
+            ops = [await self._cache_queue.get()]
+            while not self._cache_queue.empty():
+                ops.append(self._cache_queue.get_nowait())
+            # Filter ops for only what we need
+            ops = [
+                (query_req, units) for query_req, units in ops
+                if len(units) == 1
+                and units[0].cache_sql
+                and units.cache_state == CacheState.Pending
+            ]
+            if not ops:
+                continue
+
+            # TODO: Should we do any sort of error handling here?
+            g = execute.build_cache_persistence_units(ops)
+            conn = await self.tenant.acquire_pgcon(self.name)
+            try:
+                await g.execute(conn, self)
+            finally:
+                self.tenant.release_pgcon(self.name, conn)
+
+            for _, units in ops:
+                units.cache_state = CacheState.Present
+                self._cache_notify_queue.put_nowait(units[0].cache_key)
+
+    async def cache_notifier(self):
+        while True:
+            try:
+                await self._cache_notifier()
+            except Exception as ex:
+                debug.dump(ex)
+                metrics.background_errors.inc(
+                    1.0, self.tenant._instance_name, "cache_notifier"
+                )
+                # Give things time to recover, since the likely
+                # failure mode here is a failover or some such.
+                await asyncio.sleep(0.1)
+
+    async def _cache_notifier(self):
+        MAX_WAIT = 1.0
+        DELAY_AMT = 0.2
+        # 100 keys will take up about 4000 bytes, which fits in the 8000
+        # allowed in events.
+        MAX_KEYS = 100
+
+        pending_keys = []
+        last_signal = 0
+        target_time = None
+
+        while True:
+            # "Debounce" and batch query-cache-change notifications.
+            # The basic algorithm is that if a notification comes in
+            # less than DELAY_AMT since we triggered the previous one,
+            # then instead of sending it immediately, we wait an
+            # additional DELAY_AMT from then. If we are already waiting,
+            # any message also extends the wait, up to MAX_WAIT.
+            # Also, cap the batch size to MAX_KEYS, since
```

Comment:
```suggestion
            # Also, cap the batch size to MAX_KEYS, since we include the keys in the
            # Postgres event which has a size limit.
```